artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTProtocolManager.java

```diff
@@ -62,11 +73,44 @@
                        List<BaseInterceptor> outgoingInterceptors) {
       this.server = server;
       this.updateInterceptors(incomingInterceptors, outgoingInterceptors);
+      server.getManagementService().addNotificationListener(this);
    }
 
    @Override
    public void onNotification(Notification notification) {
-      // TODO handle notifications
+      if (!(notification.getType() instanceof CoreNotificationType))
+         return;
+
+      CoreNotificationType type = (CoreNotificationType) notification.getType();
+      if (type != CONSUMER_CREATED)
+         return;
+
+      TypedProperties props = notification.getProperties();
+
+      SimpleString protocolName = props.getSimpleStringProperty(ManagementHelper.HDR_PROTOCOL_NAME);
+
+      if (protocolName == null || !protocolName.toString().equals(MQTTProtocolManagerFactory.MQTT_PROTOCOL_NAME))
+         return;
+
+      int distance = props.getIntProperty(ManagementHelper.HDR_DISTANCE);
+
+      if (distance > 0) {
+         SimpleString queueName = props.getSimpleStringProperty(ManagementHelper.HDR_ROUTING_NAME);
+
+         Binding binding = server.getPostOffice().getBinding(queueName);
+         if (binding != null) {
+            Queue queue = (Queue) binding.getBindable();
+            String clientId = props.getSimpleStringProperty(ManagementHelper.HDR_CLIENT_ID).toString();
+            //If the client ID represents a client already connected to the server then the server MUST disconnect the existing client.
+            //Avoid consumers with the same client ID in the cluster appearing at different nodes at the same time
+            Collection<Consumer> consumersSet = queue.getConsumers();
```

Comment:
Might be worth adding a new method on queue that takes an additional predicate parameter, this way can pass in a predicate of 
clientId.equals(serverConsumer.getConnectionClientID()
and return only the consumers matching, this will avoid expensive copying of all consumers if the queue has high consumer count, and repetitive iterating over those.

Reply:
e.g. adding something like

```
   @Override
   public Set<Consumer> getConsumers(Predicate<Consumer> consumerPredicate) {
      Set<Consumer> consumersSet = new HashSet<>();
      for (ConsumerHolder<? extends Consumer> consumerHolder : consumers) {
         if (consumerPredicate.test(consumerHolder.consumer)) {
            consumersSet.add(consumerHolder.consumer);
         }
      }
      return consumersSet;
   }
```

Reply:
got it

Reply:
i sent this in pr to your branch